mindmeld/components/question_answerer.py

```diff
@@ -449,805 +591,2172 @@ def _generate_mapping_data(embedder_model, embedding_fields):
+                    or self.field_info.is_location_field()
+                ):
+                    raise ValueError(
+                        "Custom sort criteria can only be defined for"
+                        + " 'number', 'date' or 'location' fields."
+                    )
+
+    class FieldInfo:
+        """This class models an information source of a knowledge base field metadata"""
+
+        NUMBER_TYPES = {
+            "long",
+            "integer",
+            "short",
+            "byte",
+            "double",
+            "float",
+            "half_float",
+            "scaled_float",
+        }
+        TEXT_TYPES = {"text", "keyword"}
+        DATE_TYPES = {"date"}
+        GEO_TYPES = {"geo_point"}
+        VECTOR_TYPES = {"dense_vector"}
+
+        def __init__(self, name, field_type):
+            self.name = name
+            self.type = field_type
+
+        def get_name(self):
+            """Returns knowledge base field name"""
+
+            return self.name
+
+        def get_type(self):
+            """Returns knowledge base field type"""
+
+            return self.type
+
+        def is_number_field(self):
+            """Returns True if the knowledge base field is a number field, otherwise returns False
+            """
+
+            return self.type in self.NUMBER_TYPES
+
+        def is_date_field(self):
+            """Returns True if the knowledge base field is a date field, otherwise returns False"""
+
+            return self.type in self.DATE_TYPES
+
+        def is_location_field(self):
+            """Returns True if the knowledge base field is a location field, otherwise returns False
+            """
+
+            return self.type in self.GEO_TYPES
+
+        def is_text_field(self):
+            """Returns True if the knowledge base field is a text field, otherwise returns False"""
+
+            return self.type in self.TEXT_TYPES
+
+        def is_vector_field(self):
+            """Returns True if the knowledge base field is a vector field, otherwise returns False
+            """
+
+            return self.type in self.VECTOR_TYPES
+
+
+class NonElasticsearchQuestionAnswerer(BaseQuestionAnswerer):
+    """
+    A question answerer class not using Elasticsearch
+    """
+
+    def _get(self, index, size=10, query_type=None, app_namespace=None, **kwargs):
+
+        doc_id = kwargs.get("id")
+
+        query_type = query_type or self._query_type
+
+        # fix related to Issue 219: https://github.com/cisco/mindmeld/issues/219
+        app_namespace = app_namespace or self.app_namespace
+
+        # If an id was passed in, simply retrieve the specified document
+        if doc_id:
+            logger.info(
+                "Retrieve object from KB: index= '%s', id= '%s'.", index, doc_id
+            )
+            s = self.build_search(index, app_namespace=app_namespace)
+
+            if NonElasticsearchQuestionAnswerer.FieldResource.is_number(doc_id):
```

Comment:
We expect doc_id to be unique, however, it can be a string or a number. In either case, a filter operation that looks for exact match is possible. In case of string doc_ids, it is also possible to do a fuzzy match instead of exact match. 

For reference, please see [this docstring of filter op in Elasticsearch based QA](https://github.com/cisco/mindmeld/blob/8d21d3dd372c7ef6ac5c55025d2f9273487d9b61/mindmeld/components/question_answerer.py#L676) which mentions `If the entity is not in the canonical form, a fuzzy filter can be applied by setting the query_type to 'text'.` 

So in case of non-Es QA, this fuzzy match is not implemented irrespective of the query type.